### ISSUE TITLE
Release planning

### ISSUE THREAD

**Issue by sk1p** [LiberTEM/LiberTEM-blobfinder#7]
- [ ] first, release `libertem_blobfinder` version 0.4.0 (that's where we start, right?) with dependency on `libertem>=0.4.0.dev0'
 - [ ] then, in `libertem` master, add a dependency on `libertem_blobfinder>=0.4.0`
 - [ ] replace blobfinder code in `libertem` with imports from `libertem_blobfinder` w/ `DeprecationWarning`
 - [ ] release `libertem` 0.4.0
 - [ ] in `libertem` 0.5.0.dev0, remove dependency on `libertem_blobfinder`
 
Once this is done, the blobfinder code can evolve more freely, we can untangle the libertem-related functionality from the independent parts (see #5 for example)

**Comment by uellue** [comment]
Having parallel version numbers for all libertem-* packages is a good idea, by the way.

**Comment by sk1p** [comment]
> Having parallel version numbers for all libertem-* packages is a good idea, by the way.

For the most part I would agree. Once we extract more of the utility packages, however, it may be a good idea to let those "run" a bit faster. The leaves in the dependency tree should move faster as not to block progress in their dependents. As an example, imagine you are adding a new mask-generating function to libertem-masks, to support a new UDF in some other libertem-foo package. If we keep the versions in sync, the change in libertem-foo would have to wait for the next round of releases.

**Comment by sk1p** [comment]
What do you think about postponing #8 and #13 until after the release? The documentation will be pushed continuously anyways...

**Comment by uellue** [comment]
I was just about to work on documentation. It would mostly be copy and paste from LiberTEM. I'd prefer to spend one or two hours on getting it towards a minimum viable state before releasing.

**Comment by sk1p** [comment]
Ok :+1:

**Comment by sk1p** [comment]
Anything missing for RC? I think we are good to go!

**Comment by uellue** [comment]
I think so too. :-)

**Comment by uellue** [comment]
> Once we extract more of the utility packages, however, it may be a good idea to let those "run" a bit faster. The leaves in the dependency tree should move faster as not to block progress in their dependents. As an example, imagine you are adding a new mask-generating function to libertem-masks, to support a new UDF in some other libertem-foo package. If we keep the versions in sync, the change in libertem-foo would have to wait for the next round of releases.

Good point! Since we are working on keeping a roughly monthly release cycle for main LiberTEM, we should be able to bring things back into sync in a timely manner. We'd perhaps have a week or two where the master branch of one package would depend on a development version of another. A bit similar to the current libertem_blobfinder situation.

**Comment by uellue** [comment]
> Anything missing for RC? I think we are good to go!

Ah, I just remembered that we have to create a new record on Zenodo and update the record IDs! Otherwise we overwrite main LiberTEM. I'll take care of it!

**Comment by uellue** [comment]
> create a new record on Zenodo and update the record IDs

Done, including settings for Travis. Sandbox parent is 494128, production parent is 3663437. Looking forward to test with a RC to confirm that it works as intended.

**Comment by sk1p** [comment]
blobfinder v0.4.0 released :tada: :tada: - zenodo / github / pypi integration works fine.

One thing to think about - we may want to have semver-ish dependencies from libertem to the extracted packages - something like `libertem-blobfinder>=0.4.0,<1` - that way, if we want to break the `libertem-blobfinder` interface, we can do so by doing a major release, without affecting older libertem versions.
